src/models/events.model.js

```diff
@@ -14,16 +38,16 @@ function createModel(app) {
   const { Schema } = mongooseClient;
   const event = new Schema(
     {
-      logIndex: { type: Number, required: true },
+      logIndex: { type: Number, required: true, index: true },
       transactionIndex: { type: Number, required: true },
-      transactionHash: { type: String, required: true },
+      transactionHash: { type: String, required: true, index: true },
       blockHash: { type: String, required: true },
       blockNumber: { type: Number, required: true },
       address: { type: String, required: true },
       type: { type: String },
       id: { type: String, required: true },
       returnValues: { type: Object },
-      event: { type: String },
+      event: { type: String, index: true, enum: Object.values(EVENT_TYPES) },
```

Comment:
Does enum improve performance? what's the benefit of adding it?

Reply:
If there is no advantage, remove it. Because it may cause some error and we don't have time test it right now.

Reply:
No it wasn't related to performance however I added it for readability ( you can use distinct in production db to see the available values to see if enum is correct or not)

but I can delete it